packages/techdocs-common/src/helpers.ts

```diff
@@ -124,46 +119,74 @@ export const checkoutGitRepository = async (
   const repositoryTmpPath = await getGitRepositoryTempFolder(repoUrl);
   const token = await getTokenForGitRepo(repoUrl);
 
-  if (fs.existsSync(repositoryTmpPath)) {
-    try {
-      const repository = await Repository.open(repositoryTmpPath);
-      const currentBranchName = (
-        await repository.getCurrentBranch()
-      ).shorthand();
-      await repository.fetch('origin');
-      await repository.mergeBranches(
-        currentBranchName,
-        `origin/${currentBranchName}`,
-      );
-      return repositoryTmpPath;
-    } catch (e) {
-      logger.info(
-        `Found error "${e.message}" in cached repository "${repoUrl}" when getting latest changes. Removing cached repository.`,
-      );
-      fs.removeSync(repositoryTmpPath);
-    }
-  }
+  // Initialize a git client
+  let git = Git.fromAuth({ logger });
 
+  // Docs about why username and password are set to these specific values.
+  // https://isomorphic-git.org/docs/en/onAuth#oauth2-tokens
   if (token) {
     const type = getGitRepoType(repoUrl);
     switch (type) {
+      case 'github':
+        git = Git.fromAuth({
+          username: token,
+          password: 'x-oauth-basic',
+          logger,
+        });
+        parsedGitLocation.token = `${token}:x-oauth-basic`;
+        break;
       case 'gitlab':
-        // Personal Access Token
+        git = Git.fromAuth({
+          username: 'oauth2',
+          password: token,
+          logger,
+        });
         parsedGitLocation.token = `dummyUsername:${token}`;
         parsedGitLocation.git_suffix = true;
         break;
-      case 'github':
-        parsedGitLocation.token = `${token}:x-oauth-basic`;
+      case 'azure/api':
+        git = Git.fromAuth({
+          username: 'notempty',
+          password: token,
+          logger: logger,
+        });
```

Comment:
I feel the default values for different providers i.e. 'x-auth-token', 'oauth2', 'notempty' etc. shouldn't be here and can be done in `Git` factory method.

Reply:
I'm thinking that eventually we'd use the integrations and it would become `Git.forUrl(url)` So then the auth would be taken care of there. But we don't have that yet :)

Reply:
Yep, no worries for now!